mmengine/utils/progressbar_rich.py

```diff
@@ -0,0 +1,365 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import threading
+import time
+from collections.abc import Iterable
+from multiprocessing import Pool
+from typing import List
+
+from rich.console import Console
+from rich.progress import Progress
+
+from .timer import Timer
+
+
+class RichProgressBar:
```

Comment:
If I use `RichProgressBar` incorrectly like this:

```python
import time

from mmengine.utils import RichProgressBar

if __name__ == '__main__':
    bar = RichProgressBar()
    bar.add_single_task(total=100)
    for i in range(100):
        bar.update(1)
        time.sleep(0.1)
    del bar
```

It will raise an error like this

```
Traceback (most recent call last):
  File "/home/yehaochen/codebase/mmengine/mmengine/utils/progressbar_rich.py", line 26, in __del__
  File "/home/yehaochen/anaconda3/envs/pytorch112python38/lib/python3.8/site-packages/rich/progress.py", line 1164, in stop
  File "/home/yehaochen/anaconda3/envs/pytorch112python38/lib/python3.8/site-packages/rich/live.py", line 155, in stop
  File "/home/yehaochen/anaconda3/envs/pytorch112python38/lib/python3.8/site-packages/rich/console.py", line 1131, in line
  File "/home/yehaochen/anaconda3/envs/pytorch112python38/lib/python3.8/site-packages/rich/console.py", line 1668, in print
  File "/home/yehaochen/anaconda3/envs/pytorch112python38/lib/python3.8/site-packages/rich/console.py", line 1529, in _collect_renderables
  File "/home/yehaochen/anaconda3/envs/pytorch112python38/lib/python3.8/site-packages/rich/protocol.py", line 28, in rich_cast
ModuleNotFoundError: import of rich.console halted; None in sys.modules
```
It is a little bit hard to understood. This error is caused by the bar did not free its resources since it raised an error before, we should make sure the bar will free it resources for each exception

Reply:
Yes, I also found this problem and am improving it